src/huggingface_hub/_inference_endpoints.py

```diff
@@ -152,7 +152,10 @@ def client(self) -> InferenceClient:
                 "Cannot create a client for this Inference Endpoint as it is not yet deployed. "
                 "Please wait for the Inference Endpoint to be deployed using `endpoint.wait()` and try again."
             )
-        return InferenceClient(model=self.url, token=self._token)
+        return InferenceClient(
+            model=self.url,
+            token=self._token,  # type: ignore[arg-type] # boolean token shouldn't be possible. In practice it's ok.
```

Comment:
I've changed `InferenceClient.token` type annotation to `Optional[str]` as it simplifies a lot the logic (no need to handle bool values in external providers). In practice HF provider still supports it so it's fine to use it in that case.